### PR TITLE
DM-6500: irsaviewer conversion

### PR DESCRIPTION
**Pull request by loitly**
for details: https://jira.lsstcorp.org/browse/DM-6500

also made these changes:
remove irsa footer from fireflyviewer
filter by selected rows
auto-correct table’s filter input.
multiple columns sort via sortByCols
fix menu item not showing selected
ife automatically builds firefly.

To test, make sure you pull ife repos as well.
same branch name on ife.

### CODE REVIEW

**Comment by tgoldina** [comment]
Some issues I've noticed when testing irsaviewer:

1. We might want to set minimum width to accommodate image and catalog search panels. If the browser window is more narrow than their width, the layout is getting messed: Login link in the menu bar on top gets cut, search panels are hidden on the left, menu bar and image thumbnail are misaligned, etc.
2. If the first image search fails, there is no feedback, even in the console. A screen stays empty with a tiny square in the middle. 
3. http://localhost:8080/irsaviewer/images/fftools-logo-16x16.png Failed to load resource: the server responded with a status of 404 (Not Found)

Not directly related to this ticket (we might need separate tickets for these):

- 2MASS All-Sky 'Read Me!' link in Catalog Search panel points to http://localhost:8080/irsaviewer/Q'http://hades.ipac.caltech.edu/applications/Gator/GatorAid/irsa/scan.html'E 
- If you search m31 2MASS scan info catalog (4th catalog in the list), then bring back catalog panel, the first catalog is highlighted, but if you search, you'll notice that you are still searching 2MASS scan info.
- Both image and XY plot should handle sort action. After sorting on a column the highlight does not match the table. (Notice that when you click on an image or xy plot point, the highlight jumps on some other point.) 
- The catalog overlay points can appear beyond image limits. Test: search 2mass image for m31 with the default settings, then search for m31 catalog with the default settings. Notice catalog overlay squares on the right of the image.

**Comment by ejoliet** [comment]
The 'Read me!' link is known issue, is a bug in the server when getting 
the IRSA catalog table, on how the link of that attirbute get parsed.
Here:
in 
edu.caltech.ipac.firefly.server.catquery.CatMasterTableQuery#modifyToFullURL

Please, raise ticket and fix.
E.

On 6/14/16 10:14 AM, Tatiana Goldina wrote:
>
> Some issues I've noticed when testing irsaviewer:
>
>  1. We might want to set minimum width to accommodate image and
>     catalog search panels. If the browser window is more narrow than
>     their width, the layout is getting messed: Login link in the menu
>     bar on top gets cut, search panels are hidden on the left, menu
>     bar and image thumbnail are misaligned, etc.
>  2. If the first image search fails, there is no feedback, even in the
>     console. A screen stays empty with a tiny square in the middle.
>  3. http://localhost:8080/irsaviewer/images/fftools-logo-16x16.png
>     Failed to load resource: the server responded with a status of 404
>     (Not Found)
>
> Not directly related to this ticket (we might need separate tickets 
> for these):
>
>   * 2MASS All-Sky 'Read Me!' link in Catalog Search panel points to
>     http://localhost:8080/irsaviewer/Q'http://hades.ipac.caltech.edu/applications/Gator/GatorAid/irsa/scan.html'E
>     <http://hades.ipac.caltech.edu/applications/Gator/GatorAid/irsa/scan.html%27E>
>
>   * If you search m31 2MASS scan info catalog (4th catalog in the
>     list), then bring back catalog panel, the first catalog is
>     highlighted, but if you search, you'll notice that you are still
>     searching 2MASS scan info.
>   * Both image and XY plot should handle sort action. After sorting on
>     a column the highlight does not match the table. (Notice that when
>     you click on an image or xy plot point, the highlight jumps on
>     some other point.)
>   * The catalog overlay points can appear beyond image limits. Test:
>     search 2mass image for m31 with the default settings, then search
>     for m31 catalog with the default settings. Notice catalog overlay
>     squares on the right of the image.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/Caltech-IPAC/firefly/pull/102#issuecomment-225950717>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AAvATGJj4PC9s9oujufai_likOG8GR-Dks5qLuFZgaJpZM4I04bC>.
>

-- 
Emmanuel Joliet
NASA/IPAC  Infrared Science Archive
MS 100-22, Caltech, Pasadena, CA 91125
Office: 170
Email:  ejoliet@ipac.caltech.edu
Phone:  (626) 395-1489

File: src/firefly/js/core/AppDataCntlr.js [hunk 1]
```diff
@@ -310,7 +304,7 @@ export function dispatchOnAppReady(callback) {
 /*---------------------------- EXPORTED FUNTIONS -----------------------------*/
 export function isAppReady() {
     return get(flux.getState(), [APP_DATA_PATH, 'isReady']) &&
-           get(flux.getState(), [APP_DATA_PATH, 'gwtLoaded']);
+        (get(window, 'firefly.noGWT') || get(flux.getState(), [APP_DATA_PATH, 'gwtLoaded']));
```
Comment (tgoldina) [inline src/firefly/js/core/AppDataCntlr.js]: Just curious where firefly.noGWT is coming from.

File: src/firefly/js/tables/ui/BasicTableView.jsx [hunk 2]
```diff
@@ -234,15 +263,16 @@ function makeColumns ({columns, columnWidths, data, selectable, showUnits, showF
         );
     });
     if (selectable) {
-        var cbox = <Column
+        const checked = selectInfoCls.isSelectAll();
+        var cbox = (<Column
             key='selectable-checkbox'
             columnKey='selectable-checkbox'
-            header={<SelectableHeader checked={selectInfoCls.isSelectAll()} onSelectAll={onSelectAll} />}
+            header={<SelectableHeader {...{checked, onSelectAll, showUnits, showFilters, onFilterSelected}} />}
```
Comment (tgoldina) [inline src/firefly/js/tables/ui/BasicTableView.jsx]: I like the way you use JSX spread attributes everywhere.

**Pull request merged**